oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocument.java

```diff
@@ -36,15 +36,15 @@
     private static final Logger LOG = LoggerFactory.getLogger(ElasticDocument.class);
 
     private final String path;
-    private final List<String> fulltext;
+    private final Set<String> fulltext;
     private final Set<String> suggest;
     private final List<String> notNullProps;
     private final List<String> nullProps;
     private final Map<String, Object> properties;
 
     ElasticDocument(String path) {
         this.path = path;
-        this.fulltext = new ArrayList<>();
+        this.fulltext = new HashSet<>();
```

Comment:
What about using LinkedHashSet? Like the ArrayList, the iteration order would match the insertion order. And unlike the ArrayList, we don't have duplicate entries.

Reply:
Done. Even if from a query perspective the order is not relevant I think it is good to have a predefined order especially in cases we need to look into the content to investigate potential issues.